### PR TITLE
Fixes #7205 - Add warning about using SetSeed

### LINKED ISSUE #7205: Add note to Get-Random regarding the use of -SetSeed
# Summary of the new document or enhancement

- List of articles that need to be updated: https://docs.microsoft.com/powershell/module/microsoft.powershell.utility/get-random
- New articles that need to be created:
- Link(s) to related code PR(s) in the PowerShell/PowerShell repo:
- Link(s) to related issue(s) in the PowerShell/PowerShell repo:

## Description of what changed

From an internal email discussion. Need stronger wording about using the `-SetSeed` parameter. Using this parameter should only be used for debugging since it affects the randomness, globally, for the current session.

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1813632](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1813632) - Fixes #7205 - Add warning about using SetSeed

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [x] Preview content
- [x] Version 7.1 content
- [x] Version 7.0 content
- [x] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md [hunk 1]
```diff
@@ -35,9 +34,20 @@ The `Get-Random` cmdlet gets a randomly selected number. If you submit a collect
 Without parameters or input, a `Get-Random` command returns a randomly selected 32-bit unsigned
 integer between 0 (zero) and **Int32.MaxValue** (`0x7FFFFFFF`, `2,147,483,647`).
 
+By default, `Get-Random` generates cryptographically strong randomness using the
```
Comment (SteveL-MSFT) [inline reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md:37]:
```suggestion
By default, `Get-Random` generates cryptographically secure randomness using the
```

File: reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md [hunk 2]
```diff
@@ -35,9 +34,20 @@ The `Get-Random` cmdlet gets a randomly selected number. If you submit a collect
...
+[RandomNumberGenerator()](/dotnet/api/system.security.cryptography.randomnumbergenerator) method.
+
 You can use the parameters of `Get-Random` to specify a seed number, minimum and maximum values, and
 the number of objects returned from a submitted collection.
 
+> [!CAUTION]
+> Setting the seed results in non-random behavior. It should only be used when trying to reproduce
```
Comment (SteveL-MSFT) [inline reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md:44]:
```suggestion
> Setting the seed deliberately results in non-random repeatable behavior. It should only be used when trying to reproduce
```

File: reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md [hunk 3]
```diff
@@ -318,15 +328,16 @@ Accept wildcard characters: False
 
 ### -SetSeed
 
-Specifies a seed value for the random number generator. This seed value is used for the current
-command and for all subsequent `Get-Random` commands in the current session until you use
-**SetSeed** again or close the session. You can't reset the seed to its default value.
+Specifies a seed value for the random number generator. When you use **SetSeed**, the cmdlet uses
+the [Sytem.Random](/dotnet/api/system.random) method to generate pseudorandom number.
```
Comment (SteveL-MSFT) [inline reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md:332]:
```suggestion
the [Sytem.Random](/dotnet/api/system.random) method to generate pseudorandom number which is not cryptographically secure.
```

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
You need to apply the same change to all the versions

File: reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md [hunk 4]
```diff
@@ -361,8 +375,8 @@ submitted collection.
 
 ## NOTES
 
-`Get-Random` sets a default seed for each session based on the system time clock when the session
-starts.
+By default, `Get-Random` generates cryptographically strong randomness using the
```
Comment (SteveL-MSFT) [inline reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md:378]:
```suggestion
By default, `Get-Random` generates cryptographically secure randomness using the
```

File: reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md [hunk 5]
```diff
@@ -361,8 +375,8 @@ submitted collection.
...
+[RandomNumberGenerator()](/dotnet/api/system.security.cryptography.randomnumbergenerator) method.
```
Comment (SteveL-MSFT) [inline reference/5.1/Microsoft.PowerShell.Utility/Get-Random.md:379]:
```suggestion
[RandomNumberGenerator](/dotnet/api/system.security.cryptography.randomnumbergenerator) class.
```

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

File: reference/7.1/Microsoft.PowerShell.Utility/Get-Random.md [hunk 6]
```diff
@@ -42,8 +40,20 @@ The `Get-Random` cmdlet gets a randomly selected number. If you submit a collect
 Without parameters or input, a `Get-Random` command returns a randomly selected 32-bit unsigned
 integer between 0 (zero) and **Int32.MaxValue** (`0x7FFFFFFF`, `2,147,483,647`).
 
-You can use the parameters of `Get-Random` to specify a seed number, minimum and maximum values, the
-number of objects returned from a submitted collection, and the entire collection in a random order.
+By default, `Get-Random` generates cryptographically secure randomness using the
+[RandomNumberGenerator](/dotnet/api/system.security.cryptography.randomnumbergenerator) class.
+
+You can use the parameters of `Get-Random` to specify a seed number, minimum and maximum values, and
```
Comment (chasewilson) [inline reference/7.1/Microsoft.PowerShell.Utility/Get-Random.md:46]:
```suggestion
You can use the parameters of `Get-Random` to specify a seed number, minimum and maximum values for random selection, and
```

Comment (chasewilson) [reply reference/7.1/Microsoft.PowerShell.Utility/Get-Random.md:46]: I don't know if this is needed, it just tripped me up a little bit on the first read through.

**Review by chasewilson — APPROVED** [review]

**Pull request merged**